engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoMultiValue.java

```diff
@@ -0,0 +1,93 @@
+package org.enso.interpreter.runtime.data;
+
+import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.enso.interpreter.node.callable.resolver.MethodResolverNode;
+import org.enso.interpreter.runtime.callable.UnresolvedSymbol;
+import org.enso.interpreter.runtime.callable.function.Function;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+import org.graalvm.collections.Pair;
+
+@ExportLibrary(TypesLibrary.class)
+@ExportLibrary(InteropLibrary.class)
+public final class EnsoMultiValue implements EnsoObject {
+  @CompilationFinal(dimensions = 1)
+  private final Type[] types;
```

Comment:
Do you check anywhere for duplicates, i.e. `Text & Text & Integer`?

Reply:
No, I don't the conversion would then happen twice and the first value would be used.

Reply:
Testing what happens when multiple conversions are possible: 640578f